federation/cmd/kubefed/app/kubefed.go

```diff
@@ -17,19 +17,22 @@ limitations under the License.
 package app
 
 import (
+	"fmt"
 	"os"
 
 	"k8s.io/kubernetes/federation/pkg/kubefed"
 	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/util/logs"
+	"k8s.io/kubernetes/pkg/version"
 	_ "k8s.io/kubernetes/pkg/version/prometheus" // for version metric registration
 )
 
 func Run() error {
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
-	cmd := kubefed.NewKubeFedCommand(cmdutil.NewFactory(nil), os.Stdin, os.Stdout, os.Stderr)
+	defaultImage := fmt.Sprintf("gcr.io/google_containers/hyperkube-amd64:%s", version.Get())
```

Comment:
It's not the ImageName that changes, it's the version. I'd prefer we keep the reference to hyperkubeImageName as was done [here](https://github.com/kubernetes/kubernetes/pull/46020/files#diff-e7243d86cc50e209d78ce51d57dad48dL158). Other than that, LGTM.